src/lib_db.ml

```diff
@@ -21,6 +21,10 @@ type t =
   }
 
 let rec internal_name_scope t ~dir =
+  if not (Path.is_descendant dir ~of_:t.anonymous_root) then (
+    Fmt.failwith "scope dir %a must be a descendant of %a"
```

Comment:
I didn't know about `Fmt.failwith` but we shouldn't use this here as it raises `Failure` which is reported as an uncaught exception. I don't think the user can trigger this error with a malformed jbuild file, so instead with should use `code_errorf` or `Sexp.code_error`.

Reply:
I see. I didn't think that this would be anything other than a bug in our code.